pkg/apis/policy/v1alpha1/retry.go

```diff
@@ -0,0 +1,89 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Retry is the type used to represent a Retry policy.
+// A Retry policy authorizes retries to failed attempts for outbound traffic
+// from one service source to one or more destination services.
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type Retry struct {
+	// Object's type metadata
+	metav1.TypeMeta `json:",inline"`
+
+	// Object's metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the Retry policy specification
+	// +optional
+	Spec RetrySpec `json:"spec,omitempty"`
+}
+
+// RetrySpec is the type used to represent the Retry policy specification.
+type RetrySpec struct {
+	// Source defines the source the Retry policy applies to.
+	Source RetrySourceSpec `json:"source"`
+
+	// Destinations defines the list of destinations the Retry policy applies to.
+	Destinations []RetryDestinationSpec `json:"destinations"`
+
+	// RetryPolicy defines the retry policy the Retry policy applies.
+	RetryPolicy RetryPolicySpec `json:"retryPolicy"`
+}
+
+// RetrySourceSpec is the type used to represent the Source specified in an
+// Retry policy specification.
+type RetrySourceSpec struct {
+	// Kind defines the kind for the source in the Retry policy.
+	Kind string `json:"kind"`
+
+	// Name defines the name of the source for the given Kind.
+	Name string `json:"name"`
+
+	// Namespace defines the namespace for the given source.
+	// +optional
```

Comment:
If this is optional, what is the default ns assumed ?

Reply:
based on your crd the namespace is required, please remove the optional from here and below